x-pack/plugins/observability_solution/inventory/server/feature.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { DEFAULT_APP_CATEGORIES } from '@kbn/core/server';
+import { KibanaFeatureScope } from '@kbn/features-plugin/common';
+
+const INVENTORY_SERVER_FEATURE_ID = 'inventory';
+
+export const INVENTORY_SERVER_FEATURE = {
+  id: INVENTORY_SERVER_FEATURE_ID,
+  name: i18n.translate('xpack.inventory.featureRegistry.inventoryFeatureName', {
+    defaultMessage: 'Inventory',
+  }),
+  order: 1000,
+  category: DEFAULT_APP_CATEGORIES.observability,
+  scope: [KibanaFeatureScope.Spaces, KibanaFeatureScope.Security],
```

Comment:
We locked the features visible within a space as of `8.16`, in order to support the new solution-based navigation.
This prevents new features from appearing in Space Visibility Toggles. 

Further features registered should have only `Security` scope
```suggestion
  scope: [KibanaFeatureScope.Security],
```

Reply:
Hi @elena-shostak, 

Does this mean the idea is to remove the Space Visibility Toggles? I apologize, I’m not fully up to date with the changes being planned in that area, and I reached that conclusion based on your comment. If there’s any documentation or issue that provides more context on what you mentioned, it would be great if you could share it.

If I remove `KibanaFeatureScope.Spaces`, I encounter issues when trying to save space configurations if the new Inventory plugin we are adding isn’t selected.

![Screenshot 2024-09-26 at 14 14 09](https://github.com/user-attachments/assets/ba1e75c6-4c47-4d5c-aaf3-f23ac00d8046)

Reply:
> If there’s any documentation or issue that provides more context on what you mentioned, it would be great if you could share it.
We've added description here https://github.com/elastic/kibana/blob/main/docs/developer/architecture/security/feature-registration.asciidoc

> Does this mean the idea is to remove the Space Visibility Toggles?
Not remove, but lock the features available to it

> If I remove KibanaFeatureScope.Spaces, I encounter issues when trying to save space configurations if the new Inventory plugin we are adding isn’t selected.

yeah, this is intended

@legrego seems that this new feature needs a `space` scope, shall we also then adjust capabilites switcher logic for solution view?

Reply:
Thank you for the explanation @elena-shostak.

Do you know if there is a way to control whether the plugin appears in the configuration for both spaces and security based on an Advanced Setting? This is something we are trying to accomplish here, and I'm not sure if it's possible.

Reply:
@iblancof Sorry for the issue, we have an issue with not filtering out only `space` scope features. Raised the PR for that https://github.com/elastic/kibana/pull/194157 

Once merged you will be able to just use the `security` scope and this will be automatically filtered for spaces.

Reply:
Hey @elena-shostak, after some discussion, we decided it would be better to hide the feature in both Space Visibility Toggles and Security contexts. To achieve this, we removed the `scope` attribute from the KibanaFeatureConfig, expecting it to be hidden everywhere. However, it’s still visible in Security. Should it behave this way?

This won’t block the PR if the product team agrees to go live this way, but I wanted to check in case this is a malfunction.

Reply:
> we decided it would be better to hide the feature in both Space Visibility Toggles and Security contexts. To achieve this, we removed the scope attribute from the KibanaFeatureConfig, expecting it to be hidden everywhere. However, it’s still visible in Security. Should it behave this way?

If you don't want this exposed in Role Management, then there's no need to register the feature at all. You can simply remove the call to `registerKibanaFeature`.

This raises the question: How is access to this feature controlled? If we aren't registering an additional privilege for the Inventory feature, what other privilege will secure access to it?

Reply:
That's what I thought when I opened the PR, but without registering it as a Kibana feature, the feature is visible in all spaces, regardless of the configuration I set. That’s why I decided to register it.


![Screenshot 2024-09-27 at 13 15 12](https://github.com/user-attachments/assets/72c7b41c-9abd-4b59-8def-902b08c23a1b)

Reply:
We also attempted to use the `hidden` attribute from `KibanaFeatureConfig`, but we encountered a console error, so we decided to discard that option.

````
Error: [hidden]: definition for this key is missing

Reply:
Hey @elena-shostak, thanks for all your input and collaboration.

Given the circumstances, it seems best to proceed with both scopes defined, allowing us to move forward with the PR and use the configuration that best adapts to our current needs.

[Here](https://github.com/elastic/kibana/pull/193533#issuecomment-2379325521)’s a recap of how it would look based on the latest updates.

Since you requested changes, the PR is blocked until you approve it.

Reply:
Hey @elena-shostak, thanks for all your input and collaboration.

Given the circumstances, it seems best to proceed with both scopes defined, allowing us to move forward with the PR and use the configuration that best adapts to our current needs.

[Here](https://github.com/elastic/kibana/pull/193533#issuecomment-2379325521)’s a recap of how it would look based on the latest updates.

Since you requested changes, the PR is blocked until we reach an agreement on this 🙌🏻

Reply:
Hey @iblancof sorry for delay, let me circle back on this to the team until we move forward with both scopes defined

Reply:
> This raises the question: How is access to this feature controlled? If we aren't registering an additional privilege for the Inventory feature, what other privilege will secure access to it?

The inventory plugin will be delivered as technical preview and it will be controlled by a feature flag in advanced settings (`observability:entityCentricExperience`) . If the FF is off the plugin won't be available to the end user (neither from sidenav nor url)

Reply:
Would removing `KibanaFeatureScope.Spaces` from the scope help unblock the PR and allow your team to discuss the functionality without the pressure of this PR? (Unless you for see any security issues) 

cc @elena-shostak @legrego

Reply:
We are good to keep both `KibanaFeatureScope.Spaces` and `KibanaFeatureScope.Security` for now. Thanks

Reply:
> If we remove the `KibanaFeatureScope.Spaces` from the scope, would that unblock the PR and allow you and your team to discuss this functionality without the pressure of the current PR? (Unless you for see any security issues)


If we do that, the new feature will appear for any space configuration we set up. 
Let's maintain both scopes and proceed with merging the PR.

Thank you all for your time and dedication!